src/Type/Php/BcMathStringOrNullReturnTypeExtension.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\NullType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use function gettype;
+use function strtolower;
+use PhpParser\Node\Expr\UnaryMinus;
+
+class BcMathStringOrNullReturnTypeExtension implements \PHPStan\Type\DynamicFunctionReturnTypeExtension
+{
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return $functionReflection->getName() === 'bcdiv'
+			|| $functionReflection->getName() === 'bcmod'
+			|| $functionReflection->getName() === 'bcpowmod'
+			|| $functionReflection->getName() === 'bcsqrt';
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		$secondArgument = $functionCall->args[1]->value->value;
```

Comment:
args[1] might not exist, first need to check it with isset

Reply:
it might not even have the last `->value`, depending on the expression. You need to do `$scope->getType(functionCall->args[1]->value)` instead and then check ConstantStringType.